packages/@headlessui-react/src/components/combobox/combobox.tsx

```diff
@@ -1543,6 +1692,21 @@ function OptionFn<
     onMouseLeave: handleLeave,
   }
 
+  if (virtualItem) {
+    let localOurProps = ourProps as typeof ourProps & { style: CSSProperties }
+
+    localOurProps.style = {
+      ...localOurProps.style,
+      position: 'absolute',
+      top: 0,
+      left: 0,
+      transform: `translateY(${virtualItem.start}px)`,
```

Comment:
@RobinMalfait let's add `overflowAnchor: none` https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-anchor, sometimes browser will try to adjusts scroll position that in the end could create infinite loop.